btree/btree.c

```diff
@@ -2040,35 +2041,68 @@ static void generic_move(struct nd *src, struct nd *tgt,
 }
 
 
+int  m0_btree_open(void *addr, int nob, struct m0_btree **out)
+{
+	return 0;
+}
+
+void m0_btree_close(struct m0_btree *arbor)
+{
+}
+
+void m0_btree_create(void *addr, int nob, const struct m0_btree_type *bt,
+		     struct m0_be_tx *tx, struct m0_btree_op *bop)
+{
+}
+
+void m0_btree_destroy(struct m0_btree *arbor, struct m0_btree_op *bop)
+{
+}
+
+void m0_btree_get(struct m0_btree *arbor, const struct m0_btree_key *key,
+		  const struct m0_btree_cb *cb, uint64_t flags,
+		  struct m0_btree_op *bop)
+{
+}
+
+void m0_btree_nxt(struct m0_btree *arbor, const struct m0_btree_key *key,
+		  const struct m0_btree_cb *cb, uint64_t flags,
+		  struct m0_btree_op *bop)
+{
+}
+
+void m0_btree_put(struct m0_btree *arbor, struct m0_be_tx *tx,
+		  const struct m0_btree_key *key,
+		  const struct m0_btree_cb *cb, uint64_t flags,
+		  struct m0_btree_op *bop)
+{
+}
+
 #ifndef __KERNEL__
+
+
+/**
+ *  --------------------------
+ *  Section START - Unit Tests
+ *  --------------------------
+ */
+
 /**
  * The code contained below is 'ut'. This is a little experiment to contain the
  * ut code in the same file containing the functionality code. We are open to
  * changes iff enough reasons are found that this model either does not work or
  * is not intuitive or maintainable.
  */
-static void m0_btree_ut_node_create_delete(void);
-static void m0_btree_ut_node_add_del_rec(void);
 
+#define m0_be_tx_init(tx,tid,dom,sm_group,persistent,discarded,filler,datum) \
```

Comment:
spacing in function parameters.

Reply:
There cannot be spaces here since this is a macro definition.

Reply:
oh yes, my bad.

Reply:
Why is this needed?
I guess, we want some stub functions defined. 

But, we need to undefine these macros after use.

Reply:
space is also needed in macro.